RadClass/RadClass.py

```diff
@@ -75,7 +88,10 @@ def queue_file(self):
         self.processor.init_database(self.filename, self.datapath)
         # parameters for keeping track of progress through a file
 
-        self.working_time = self.processor.timestamps[0]
+        if self.start_time is not None:
+            self.working_time = self.processor.timestamps[self.processor.timestamps > self.start_time][0]
```

Comment:
Doesn’t `start_I` depend on the start time? … and won’t you need to advance the cache?

Reply:
Correct! I have updated the `queue_file` if-else statement to update `start_i` when a `start_time` is provided using `np.where`. Since `start_i` is updated in `queue_file` at the beginning, `run_cache` should be updated as expected.